buildengine/build_go_test.go

```diff
@@ -245,3 +245,74 @@ func TestGoModVersion(t *testing.T) {
 	}
 	testBuild(t, bctx, fmt.Sprintf("go version %q is not recent enough for this module, needs minimum version \"9000.1.1\"", runtime.Version()[2:]), []assertion{})
 }
+
+func TestGeneratedTypeRegistry(t *testing.T) {
+	sch := &schema.Schema{
+		Modules: []*schema.Module{
+			{Name: "another", Decls: []schema.Decl{
+				&schema.Enum{
+					Name:   "TypeEnum",
+					Export: true,
+					Variants: []*schema.EnumVariant{
+						{Name: "A", Value: &schema.TypeValue{Value: &schema.Int{}}},
+						{Name: "B", Value: &schema.TypeValue{Value: &schema.String{}}},
+					},
+				},
+			}},
+		},
+	}
+	expected := `// Code generated by FTL. DO NOT EDIT.
+package main
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/TBD54566975/ftl/common/plugin"
+	"github.com/TBD54566975/ftl/go-runtime/server"
+	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+
+	"ftl/other"
+	"ftl/another"
+)
+
+func main() {
+	verbConstructor := server.NewUserVerbServer("other",
+		server.HandleCall(other.Echo),
+	)
+	ctx := context.Background()
+	typeRegistry := ftl.NewTypeRegistry()
```

Comment:
Can we put this in a sub-package of `go-runtime` instead of `ftl`? The `ftl` package is specifically intended for end users to import, and these aren't really user facing IIUC.

Reply:
yeah, actually working up a significant refactor with changes that emerged while implementing ingress encoding

planning to move `type_registry.go` into the schema package and expose it as a proto for other runtimes. it'll look like this:
```
type TypeRegistry struct {
	GoTypes map[string]reflect.Type `protobuf:"-"` // not encoded to proto/only used in Go
	SchemaTypes map[string]Type `protobuf:"1"`
	SumTypes map[string]SumTypeVariants `protobuf:"2"`
}
```
and we'll push these both into the context:
```
	ctx = context.WithValue(ctx, "typeRegistry", typeRegistry.ToProto())
	ctx = context.WithValue(ctx, "typeRegistryGo", typeRegistry)
```

other runtimes will need to supply `typeRegistry` in the context as well, and we'll use it to associate the variant name with a schema type. we'll inspect this in the ingress encoder

`typeRegistryGo` will only be inspected in the go-runtime encoder- we'll use it to associate variant name with `reflect.Type`

Reply:
Intuitively it feels a bit strange that `GoTypes map[string]reflect.Type` is in that struct... it feels like that's coupling something `go-runtime`-specific to the general purpose `schema` package, which shouldn't be happening?

Reply:
ah yeah that is fair, i'll split out the go specific stuff into a separate registry in `go-runtime` and keep the runtime-agnostic type registry in `backend/schema`